datalad/metadata/aggregate.py

```diff
@@ -12,31 +12,434 @@
+        # place metadata object into the source dataset
+        objpath = opj(dest.path, dirname(agginfo_relpath), objrelpath)
+
+        # write obj files
+        if exists(objpath):
+            dest.unlock(objpath)
+        # TODO actually dump a compressed file when annexing is possible
+        # to speed up on-demand access
+        jsondump(meta, objpath)
+        # stage for dataset.save()
+        to_save.append(dict(path=objpath, type='file'))
+
+        # important to use abspath here, needs to be rewritten relative to
+        # all receiving datasets
+        agginfo['{}_info'.format(mtype)] = objpath
+
+    return errored
+
+
+def _adj2subtrees(base, adj):
+    # given a set of parent-child mapping, compute a mapping of each parent
+    # to all its (grand)children of any depth level
+    subtrees = dict(adj)
+    subs = set()
+    # from bottom up
+    for ds in sorted(adj, reverse=True):
+        subtree = []
+        for sub in subtrees[ds]:
+            subs.add(sub)
+            subtree.append(sub)
+            subtree.extend(subtrees.get(sub, []))
+        subtrees[ds] = subtree
+    # give each leaf dataset an entry too
+    for sub in subs:
+        if sub not in subtrees:
+            subtrees[sub] = []
+    return subtrees
+
 
-def _store_json(ds, path, meta):
-    if not exists(path):
-        os.makedirs(path)
-    fname = opj(path, metadata_filename)
-    jsondump(meta, fname)
-    # stage potential changes
-    ds.repo.add(fname, git=True)
+def _get_latest_refcommit(ds, subds_relpaths):
+    """Find the latest commit that changed any real content
+
+    This will ignore anything at or underneath:
+
+    - .git
+    - .datalad
+    - .gitmodules
+    - .gitattributes
+    - any submodule
+
+    Returns
+    -------
+    str or None
+      None is return if no commit can be found, or no relevant content
+      files were found at all. Otherwise the full commit hash if the
+      last commit that touch any relevant content is returned.
+    """
+
+    def _filterpaths(basepath, paths, exclude):
+        final_paths = []
+        for rp in [opj(basepath, p) if basepath else p for p in paths]:
+            if rp in exclude:
+                continue
+            elif any(ep.startswith(_with_sep(rp)) for ep in exclude):
+                final_paths.extend(
+                    _filterpaths(rp, listdir(opj(ds.path, rp)), exclude))
+                pass
+            else:
+                final_paths.append(rp)
+        return final_paths
+
+    relevant_paths = _filterpaths(
+        '',
+        listdir(ds.path),
+        # NOTE: this will also ignore datalad's native dataset-global metadata
+        # rationale: the metadata still describes the dataset content, so
+        # even if it changes, the description changes, but not the content
+        # it is describing -> ref commit should be unaffected
+        list(exclude_from_metadata) + subds_relpaths)
+
+    if not relevant_paths:
+        return None
+
+    try:
```

Comment:
I guess the entire `try ... except` should be a method of `GitRepo` (`get_last_commit_from_files()` or sth)

Reply:
Make an issue and I'll assign yourself ;-)

Reply:
Created an issue. And assigned myself, of course ;-)